### PR TITLE
when creating a PR, if remote branch unavailable, try pushing it up

### PR DESCRIPTION
**Pull request by maiamcc**
when creating a PR, if remote branch unavailable, try pushing it up before giving up

### CODE REVIEW

File: git-pull-request [hunk 1]
```diff
@@ -50,7 +50,22 @@ repo.git.remote('prune', 'origin')
 try:
   diff = repo.git.diff(local_branch, remote_ref)
 except git.exc.GitCommandError:
-  util.fatal('Remote branch %s not available.' % remote_ref)
+  push_it = util.prompt('Remote branch %s not available; attempt to push it up? [Y/n]' % remote_ref)
+  if push_it.lower() in ['y', '']:
```
Comment (dpup) [inline git-pull-request]: elsewhere we've allowed `yes` perhaps we should add a util for this.

File: git-pull-request [hunk 2]
```diff
@@ -50,7 +50,22 @@ repo.git.remote('prune', 'origin')
...
+    if len(split_remote_ref) != 2:
+      util.fatal('Remote ref name "%s" in unexpected format (expected '
+        '"[origin]/[branchname]"' % remote_ref)
+      exit(1)
+    repo.git.push([split_remote_ref[0], split_remote_ref[1]])
+    try:
+      # Try again to diff against remote branch
+      diff = repo.git.diff(local_branch, remote_ref)
+    except git.exc.GitCommandError:
+      util.info('Remote branch %s is STILL not available, nothing to do.' % remote_ref)
```
Comment (dpup) [inline git-pull-request]: Should we just exit here without falling through to the error on 71?

File: git-pull-request (hunk 1 above)
Comment (maiamcc) [reply git-pull-request]: cool, done

File: git-pull-request (hunk 2 above)
Comment (maiamcc) [reply git-pull-request]: good call, fixed

File: git-sync-all [hunk 3]
```diff
@@ -14,8 +14,8 @@ import os
 import git
 import util
 
-go = util.prompt('Are you absolutely sure? [y/N]')
-if go != 'Y' and go != 'y' and go != 'yes':
+go = util.prompt_y_n('Are you absolutely sure?')
+if not go:
 	util.warn('Ok, nothing to see here...')
```
Comment (maiamcc) [inline git-sync-all]: So i preserved the current behavior and only changed ln 17 to use the new util... but it seems to me that if the user answers "no", we don't actually stop execution? bug?

Comment (dpup) [reply git-sync-all]: sounds like. ☹️ Mind fixing? Thanks!

And sorry for the delay on these reviews.

Comment (maiamcc) [reply git-sync-all]: No worries -- and fixed!

**Comment by dpup** [comment]
Thanks!

**Pull request merged**